File: 01_Language/01_Functions/python/mkdir.py
# coding: utf-8

import os


def mkdir(pathname, mode=0o777, recursive=False):
    try:
        if recursive:
            os.makedirs(pathname, mode)
        else:
            os.mkdir(pathname, mode)
        return True
    except:
        return False


if __name__ == '__main__':
    print(mkdir('./depth1/depth2/depth3/', 0o777, True))

File: 01_Language/01_Functions/go/src/functions/mkdir.go
package functions

import "os"

func Mkdir(pathname string, mode os.FileMode, recursive bool) bool {
	var err error
	if recursive {
		err = os.MkdirAll(pathname, mode)
	} else {
		err = os.Mkdir(pathname, mode)
	}
	return err == nil
}
